// @tryvital/vital-node/dist/api/resources/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.insurance = exports.testkit = exports.labTests = exports.introspect = exports.providers = exports.team = exports.user = exports.vitals = exports.meal = exports.body = exports.sleep = exports.workouts = exports.activity = exports.devices = exports.profile = exports.link = void 0;
exports.link = __importStar(require("./link"));
exports.profile = __importStar(require("./profile"));
exports.devices = __importStar(require("./devices"));
exports.activity = __importStar(require("./activity"));
exports.workouts = __importStar(require("./workouts"));
exports.sleep = __importStar(require("./sleep"));
exports.body = __importStar(require("./body"));
exports.meal = __importStar(require("./meal"));
exports.vitals = __importStar(require("./vitals"));
exports.user = __importStar(require("./user"));
exports.team = __importStar(require("./team"));
exports.providers = __importStar(require("./providers"));
exports.introspect = __importStar(require("./introspect"));
exports.labTests = __importStar(require("./labTests"));
exports.testkit = __importStar(require("./testkit"));
exports.insurance = __importStar(require("./insurance"));
__exportStar(require("./link/client/requests"), exports);
__exportStar(require("./profile/client/requests"), exports);
__exportStar(require("./devices/client/requests"), exports);
__exportStar(require("./activity/client/requests"), exports);
__exportStar(require("./workouts/client/requests"), exports);
__exportStar(require("./sleep/client/requests"), exports);
__exportStar(require("./body/client/requests"), exports);
__exportStar(require("./meal/client/requests"), exports);
__exportStar(require("./vitals/client/requests"), exports);
__exportStar(require("./user/client/requests"), exports);
__exportStar(require("./team/client/requests"), exports);
__exportStar(require("./introspect/client/requests"), exports);
__exportStar(require("./labTests/client/requests"), exports);
__exportStar(require("./testkit/client/requests"), exports);
__exportStar(require("./insurance/client/requests"), exports);
