// @tryvital/vital-node/serialization/types/ClientFacingActivity.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientFacingActivity = void 0;
const core = __importStar(require("../../core"));
exports.ClientFacingActivity = core.serialization.object({
    userId: core.serialization.property("user_id", core.serialization.string()),
    id: core.serialization.string(),
    date: core.serialization.date(),
    calendarDate: core.serialization.property("calendar_date", core.serialization.string()),
    caloriesTotal: core.serialization.property("calories_total", core.serialization.number().optional()),
    caloriesActive: core.serialization.property("calories_active", core.serialization.number().optional()),
    steps: core.serialization.number().optional(),
    dailyMovement: core.serialization.property("daily_movement", core.serialization.number().optional()),
    distance: core.serialization.number().optional(),
    low: core.serialization.number().optional(),
    medium: core.serialization.number().optional(),
    high: core.serialization.number().optional(),
    source: core.serialization.lazyObject(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).ClientFacingSource; })),
    floorsClimbed: core.serialization.property("floors_climbed", core.serialization.number().optional()),
    timeZone: core.serialization.property("time_zone", core.serialization.string().optional()),
    timezoneOffset: core.serialization.property("timezone_offset", core.serialization.number().optional()),
    heartRate: core.serialization.property("heart_rate", core.serialization.lazyObject(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).ClientFacingHeartRate; })).optional()),
});
