// @tryvital/vital-node/api/types/ClientFacingResource.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientFacingResource = void 0;
exports.ClientFacingResource = {
    Profile: "profile",
    Activity: "activity",
    Sleep: "sleep",
    Body: "body",
    Workouts: "workouts",
    WorkoutStream: "workout_stream",
    Connection: "connection",
    Order: "order",
    Result: "result",
    Appointment: "appointment",
    Glucose: "glucose",
    Heartrate: "heartrate",
    Hrv: "hrv",
    Hypnogram: "hypnogram",
    Ige: "ige",
    Igg: "igg",
    BloodOxygen: "blood_oxygen",
    BloodPressure: "blood_pressure",
    Cholesterol: "cholesterol",
    Device: "device",
    Weight: "weight",
    Fat: "fat",
    Meal: "meal",
    Water: "water",
    Caffeine: "caffeine",
    MindfulnessMinutes: "mindfulness_minutes",
    Steps: "steps",
    CaloriesActive: "calories_active",
    Distance: "distance",
    FloorsClimbed: "floors_climbed",
    RespiratoryRate: "respiratory_rate",
    Vo2Max: "vo2_max",
    CaloriesBasal: "calories_basal",
    StressLevel: "stress_level",
    ElectrocardiogramVoltage: "electrocardiogram_voltage",
    SleepStream: "sleep_stream",
};
