// @tryvital/vital-node/dist/serialization/types/ClientFacingTeam.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientFacingTeam = void 0;
const core = __importStar(require("../../core"));
exports.ClientFacingTeam = core.serialization.object({
    id: core.serialization.string(),
    orgId: core.serialization.property("org_id", core.serialization.string().optional()),
    name: core.serialization.string(),
    svixAppId: core.serialization.property("svix_app_id", core.serialization.string().optional()),
    clientId: core.serialization.property("client_id", core.serialization.string().optional()),
    clientSecret: core.serialization.property("client_secret", core.serialization.string().optional()),
    airtableApiKey: core.serialization.property("airtable_api_key", core.serialization.string().optional()),
    airtableBaseId: core.serialization.property("airtable_base_id", core.serialization.string().optional()),
    webhookSecret: core.serialization.property("webhook_secret", core.serialization.string().optional()),
    apiKey: core.serialization.property("api_key", core.serialization.string().optional()),
    apiKeys: core.serialization.property("api_keys", core.serialization
        .list(core.serialization.lazyObject(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).ClientFacingApiKey; })))
        .optional()),
    configuration: core.serialization.lazyObject(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).TeamConfig; })).optional(),
    testkitsTextsEnabled: core.serialization.property("testkits_texts_enabled", core.serialization.boolean()),
    labTestsPatientCommunicationEnabled: core.serialization.property("lab_tests_patient_communication_enabled", core.serialization.boolean()),
    labTestsPatientSmsCommunicationEnabled: core.serialization.property("lab_tests_patient_sms_communication_enabled", core.serialization.boolean()),
    labTestsPatientEmailCommunicationEnabled: core.serialization.property("lab_tests_patient_email_communication_enabled", core.serialization.boolean()),
    logoUrl: core.serialization.property("logo_url", core.serialization.string().optional()),
    delegatedFlow: core.serialization.property("delegated_flow", core.serialization.lazy(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).DelegatedFlowType; }))),
});
