// @tryvital/vital-node/dist/serialization/types/ClientFacingSleep.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientFacingSleep = void 0;
const core = __importStar(require("../../core"));
exports.ClientFacingSleep = core.serialization.object({
    userId: core.serialization.property("user_id", core.serialization.string()),
    id: core.serialization.string(),
    date: core.serialization.date(),
    calendarDate: core.serialization.property("calendar_date", core.serialization.string()),
    bedtimeStart: core.serialization.property("bedtime_start", core.serialization.date()),
    bedtimeStop: core.serialization.property("bedtime_stop", core.serialization.date()),
    timezoneOffset: core.serialization.property("timezone_offset", core.serialization.number().optional()),
    duration: core.serialization.number(),
    total: core.serialization.number(),
    awake: core.serialization.number(),
    light: core.serialization.number(),
    rem: core.serialization.number(),
    deep: core.serialization.number(),
    score: core.serialization.number().optional(),
    hrLowest: core.serialization.property("hr_lowest", core.serialization.number().optional()),
    hrAverage: core.serialization.property("hr_average", core.serialization.number().optional()),
    efficiency: core.serialization.number().optional(),
    latency: core.serialization.number().optional(),
    temperatureDelta: core.serialization.property("temperature_delta", core.serialization.number().optional()),
    skinTemperature: core.serialization.property("skin_temperature", core.serialization.number().optional()),
    hrDip: core.serialization.property("hr_dip", core.serialization.number().optional()),
    averageHrv: core.serialization.property("average_hrv", core.serialization.number().optional()),
    respiratoryRate: core.serialization.property("respiratory_rate", core.serialization.number().optional()),
    source: core.serialization.lazyObject(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).ClientFacingSource; })),
    sleepStream: core.serialization.property("sleep_stream", core.serialization.lazyObject(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).ClientFacingSleepStream; })).optional()),
});
