// @tryvital/vital-node/dist/serialization/types/HealthInsuranceCreateRequest.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.HealthInsuranceCreateRequest = void 0;
const core = __importStar(require("../../core"));
exports.HealthInsuranceCreateRequest = core.serialization.object({
    frontImage: core.serialization.property("front_image", core.serialization.lazy(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).HealthInsuranceCreateRequestFrontImage; })).optional()),
    backImage: core.serialization.property("back_image", core.serialization.lazy(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).HealthInsuranceCreateRequestBackImage; })).optional()),
    patientSignatureImage: core.serialization.property("patient_signature_image", core.serialization
        .lazy(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).HealthInsuranceCreateRequestPatientSignatureImage; }))
        .optional()),
    subjective: core.serialization.string().optional(),
    assessmentPlan: core.serialization.property("assessment_plan", core.serialization.string().optional()),
    payorCode: core.serialization.property("payor_code", core.serialization.string().optional()),
    insuranceId: core.serialization.property("insurance_id", core.serialization.string().optional()),
    responsibleRelationship: core.serialization.property("responsible_relationship", core.serialization.lazy(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).ResponsibleRelationship; })).optional()),
    responsibleDetails: core.serialization.property("responsible_details", core.serialization.lazyObject(() => __awaiter(void 0, void 0, void 0, function* () { return (yield Promise.resolve().then(() => __importStar(require("..")))).PersonDetails; })).optional()),
    diagnosisCodes: core.serialization.property("diagnosis_codes", core.serialization.list(core.serialization.string()).optional()),
});
