// @tryvital/vital-node/dist/serialization/types/LabResultsMetadata.js
"use strict";
/**
 * This file was auto-generated by Fern from our API Definition.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LabResultsMetadata = void 0;
const core = __importStar(require("../../core"));
exports.LabResultsMetadata = core.serialization.object({
    age: core.serialization.string(),
    dob: core.serialization.string(),
    clia: core.serialization.property("clia_#", core.serialization.string().optional()),
    patient: core.serialization.string(),
    provider: core.serialization.string().optional(),
    laboratory: core.serialization.string().optional(),
    dateReported: core.serialization.property("date_reported", core.serialization.string()),
    dateCollected: core.serialization.property("date_collected", core.serialization.string().optional()),
    specimenNumber: core.serialization.property("specimen_number", core.serialization.string()),
    dateReceived: core.serialization.property("date_received", core.serialization.string().optional()),
    status: core.serialization.string().optional(),
    interpretation: core.serialization.string().optional(),
});
