// @tryvital/vital-node/dist/serialization/resources/vitals/client/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.bloodPressure = exports.electrocardiogramVoltage = exports.bloodOxygen = exports.bodyFat = exports.bodyWeight = exports.cholesterol = exports.cholesterolHdl = exports.cholesterolLdl = exports.cholesterolTotal = exports.cholesterolTriglycerides = exports.glucose = exports.heartrate = exports.hrv = exports.hypnogram = exports.igg = exports.ige = exports.respiratoryRate = exports.caloriesActive = exports.caloriesBasal = exports.distance = exports.floorsClimbed = exports.steps = exports.water = exports.caffeine = exports.mindfulnessMinutes = exports.stressLevel = exports.vo2Max = void 0;
exports.vo2Max = __importStar(require("./vo2Max"));
exports.stressLevel = __importStar(require("./stressLevel"));
exports.mindfulnessMinutes = __importStar(require("./mindfulnessMinutes"));
exports.caffeine = __importStar(require("./caffeine"));
exports.water = __importStar(require("./water"));
exports.steps = __importStar(require("./steps"));
exports.floorsClimbed = __importStar(require("./floorsClimbed"));
exports.distance = __importStar(require("./distance"));
exports.caloriesBasal = __importStar(require("./caloriesBasal"));
exports.caloriesActive = __importStar(require("./caloriesActive"));
exports.respiratoryRate = __importStar(require("./respiratoryRate"));
exports.ige = __importStar(require("./ige"));
exports.igg = __importStar(require("./igg"));
exports.hypnogram = __importStar(require("./hypnogram"));
exports.hrv = __importStar(require("./hrv"));
exports.heartrate = __importStar(require("./heartrate"));
exports.glucose = __importStar(require("./glucose"));
exports.cholesterolTriglycerides = __importStar(require("./cholesterolTriglycerides"));
exports.cholesterolTotal = __importStar(require("./cholesterolTotal"));
exports.cholesterolLdl = __importStar(require("./cholesterolLdl"));
exports.cholesterolHdl = __importStar(require("./cholesterolHdl"));
exports.cholesterol = __importStar(require("./cholesterol"));
exports.bodyWeight = __importStar(require("./bodyWeight"));
exports.bodyFat = __importStar(require("./bodyFat"));
exports.bloodOxygen = __importStar(require("./bloodOxygen"));
exports.electrocardiogramVoltage = __importStar(require("./electrocardiogramVoltage"));
exports.bloodPressure = __importStar(require("./bloodPressure"));
